### PR TITLE
Suggest modules from one module

### LINKED ISSUE #112: Suggest modules stemming from a single module
## Description

By clicking on a module’s node in the graph, the user will get recommended what modules to take next, based on the importance of completing a module, and their degree requirements. Core modules would have the highest priority, and within core modules, the one that unlocks more modules will have a higher priority. For example, having completed CS1101S, a possible priority ranking would be CS2030S > CS2100 > CS2107. Here, CS2107 is ranked the lowest among the three because it is not a core module, unlike CS2030S and CS2100. Next, CS2030S is ranked higher because it unlocks more modules than CS2100.

Additionally, this feature should only suggest modules for which the user has cleared other pre-requisites. Having cleared just CS1101S for example, by clicking CS1101S, the user will not get CS2040S as a suggestion, as it requires CS1231S to be completed.

### PR DESCRIPTION
**Pull request by weiseng18**
### Summary of changes

Will close #112.

The priority for suggesting modules is currently:

1. Is it in your degree?
2. Number of modules you will unlock, after you finish this module
3. Tie break by module code lexicographical order

The `no-optional-params` test variants are the same as the original tests. The `add-module-codes` test variants init an empty user instead, and supply the optional param `addModuleCodes` to the `User` repository functions.

### CODE REVIEW

File: packages/database/src/repository/User.ts [hunk 1]
```diff
@@ -119,9 +120,11 @@ export function UserRepository(database?: DataSource): Repository {
     // 2. get array of module codes of post-reqs (fulfillRequirements)
     const postReqCodesSet = new Set<string>()
     user.modulesDone.forEach((module: Module) => {
-      module.fulfillRequirements.forEach((moduleCode: string) => {
-        postReqCodesSet.add(moduleCode)
-      })
+      // can be empty string
+      if (module.fulfillRequirements.length > 0)
```
Comment (nguyenvukhang) [inline packages/database/src/repository/User.ts:124]: use Array check

**Review by nguyenvukhang — APPROVED** [review]
